### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by ynadeem**

### CODE REVIEW

File: components/layout.js [hunk 1]
```diff
@@ -9,6 +10,9 @@ export default function Layout({ children }) {
         <div className="bg-white shadow-md my-4 mx-2 rounded w-full">
           <Navbar />
         </div>
+        {/* <div className="bg-white shadow-md my-4 mx-2 rounded w-full">
```
Comment (sbeleidy) [inline components/layout.js:13]: Can you delete this?

File: components/navbar.js [hunk 2]
```diff
@@ -1,31 +1,24 @@
...
 import Image from "next/image";
 import Link from "next/link";
+import SearchForm from "./searchForm";
 import amplitude from "amplitude-js";
+import firebaseApp from "../firebase";
+import logo from "../public/peerbots_logo.png";
+import { useFirebaseAuth } from "../auth";
 
 export default function Navbar() {
   const user = useFirebaseAuth();
```
Comment (sbeleidy) [inline components/navbar.js:17]: Can you set this to `const {user} = useFirebaseAuth();`?

File: components/navbar.js [hunk 3]
```diff
@@ -41,11 +34,40 @@ export default function Navbar() {
   }
 
   useEffect(() => {
-    if (user) {
+    if (user.user) {
```
Comment (sbeleidy) [inline components/navbar.js:37]: This can then be just user

File: components/searchForm.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  const submitSearchQuery = (e) => {
+    e.preventDefault();
+    router.push({
+      pathname: "/search",
+      query: { q: searchQueryRef.current.value },
+    });
+  };
+
+  return (
+    <>
```
Comment (sbeleidy) [inline components/searchForm.js:16]: Can you remove the fragment since we only have one div here?

File: components/oldNavbar.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
+import AuthForm from "./authForm";
```
Comment (sbeleidy) [inline components/oldNavbar.js:1]: Can we delete this file?

File: components/layout.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import Navbar from "./navbar";
 import Footer from "./footer";
 import { FirebaseAuthProvider } from "../auth";
+// import NewNavbar from "../components/newNavbar";
```
Comment (sbeleidy) [inline components/layout.js:4]: Can you also delete this?

File: components/navbar.js [hunk 7]
```diff
@@ -60,96 +82,228 @@ export default function Navbar() {
...
+                  </div>
+                )}
+                <div className="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start">
+                  <div className="flex-shrink-0 flex items-center">
+                    {user.user ? (
+                      <div>
+                        <div className="block lg:hidden w-24">
+                          <Link href="/">
+                            <a>
+                              <Image src={logo} alt="Peerbots" />
```
Comment (sbeleidy) [inline components/navbar.js:130]: Still need to figure out image src on `/my/pages`

File: components/navbar.js [hunk 8]
```diff
@@ -60,96 +82,228 @@ export default function Navbar() {
...
+                      <div></div>
+                    )}
+                  </div>
+                  <div className="hidden sm:block sm:ml-6">
+                    <div className="flex space-x-4">
+                      {navigation.map((item) => (
+                        <a
+                          key={item.name}
+                          href={item.href}
+                          className={classNames(
```
Comment (sbeleidy) [inline components/navbar.js:152]: Is this whole classnames thing how we have to do this?

File: components/navbar.js [hunk 9]
```diff
@@ -60,96 +82,228 @@ export default function Navbar() {
...
+                    <Menu
+                      as="div"
+                      className="border border-neutral-200 shadow-lg p-2 rounded-lg ml-3 relative"
                     >
-                      Sign Out
-                    </button>
+                      <div className="flex items-center">
+                        <Menu.Button className="flex items-center text-sm focus:outline-none">
+                          <span className="sr-only">Open user menu</span>
+                          <>
```
Comment (sbeleidy) [inline components/navbar.js:178]: Do we need this?

File: components/navbar.js [hunk 10]
```diff
@@ -60,96 +82,228 @@ export default function Navbar() {
...
+                        leave="transition ease-in duration-75"
+                        leaveFrom="transform opacity-100 scale-100"
+                        leaveTo="transform opacity-0 scale-95"
+                      >
+                        <Menu.Items className="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg py-1 bg-white ring-1 ring-black ring-opacity-5 focus:outline-none">
+                          <Menu.Item>
+                            {({ active }) => (
+                              <Link
+                                href="/[userId]"
+                                as={`/${user.userInDb.id}`}
```
Comment (sbeleidy) [inline components/navbar.js:225]: I guess above you should use `{ user, userInDb} = useFirebaseAuth()`

File: components/navbar.js [hunk 11]
```diff
@@ -60,96 +82,228 @@ export default function Navbar() {
...
+                    </Menu>
                   </div>
-                </>
-              ) : (
-                <>
-                  <div className="">
-                    <span>
-                      <a
+                ) : (
+                  <>
```
Comment (sbeleidy) [inline components/navbar.js:258]: We have one div here so this fragment can go as well.

File: pages/index.js [hunk 12]
```diff
@@ -20,6 +22,11 @@ export default function Home() {
         <link rel="icon" href="/favicon.ico" />
       </Head>
       <div>
+        <div className="lg:hidden">
+          <ContentRow content={[]}>
```
Comment (sbeleidy) [inline pages/index.js:26]: This should not be a content row. Content rows are to display content cards.

File: pages/index.js [hunk 13]
```diff
@@ -3,6 +3,8 @@ import LatestContentRow from "../components/latestContentRow";
 import TrustedContentRow from "../components/trustedContentRow";
 import { useEffect } from "react";
 import amplitude from "amplitude-js";
+import SearchForm from "../components/searchForm";
+import ContentRow from "../components/contentRow";
```
Comment (sbeleidy) [inline pages/index.js:7]: And I guess that means delete the import too

File: styles/globals.css [hunk 14]
```diff
@@ -4,7 +4,7 @@
 
 @layer components {
   .input-base {
-    @apply bg-primary bg-opacity-5 border-0 border-b-2 border-gray-300 focus:border-gray-600 py-2 px-2;
+    @apply bg-primary bg-opacity-5 border-2 border-gray-300 focus:border-gray-600 py-2 px-2;
```
Comment (sbeleidy) [inline styles/globals.css:7]: I think we should change this back based on the initial design for the auth form. If you think the search should be different then use this class and then add additional styling. Changing the class has also changed the auth form styles and I don't think that looks as good anymore.

File: package.json [hunk 15]
```diff
@@ -9,6 +9,7 @@
   },
   "dependencies": {
     "@headlessui/react": "^1.4.2",
+    "@heroicons/react": "^1.0.5",
```
Comment (sbeleidy) [inline package.json:12]: I've been using these without importing them. Do you prefer the SVG copy and paste or are you liking using the import from this library better? Based on your experience with it let's apply that to all our svgs

**Review by sbeleidy — CHANGES_REQUESTED** [review]
Awesome work overall. I think it'll need some more changes before we merge it though.

File: components/navbar.js (hunk 8 above)
Comment (ynadeem) [reply components/navbar.js:152]: What's wrong with it?

File: components/navbar.js (hunk 9 above)
Comment (ynadeem) [reply components/navbar.js:178]: It complains without it

File: package.json (hunk 15 above)
Comment (ynadeem) [reply package.json:12]: I think the import is more readable but I don't have a strong preference either way.

File: components/navbar.js (hunk 7 above)
Comment (ynadeem) [reply components/navbar.js:130]: This will have to be tackled in another issue. #52 could be a good place.

**Comment by sbeleidy** [comment]
Looks good to me. Fantastic work!

**Pull request merged**